model/link_metadata.go

```diff
@@ -17,10 +19,82 @@ const (
 	LINK_METADATA_TYPE_IMAGE     LinkMetadataType = "image"
 	LINK_METADATA_TYPE_NONE      LinkMetadataType = "none"
 	LINK_METADATA_TYPE_OPENGRAPH LinkMetadataType = "opengraph"
+	MAX_IMAGES                   int              = 5
 )
 
 type LinkMetadataType string
 
+// checks if original string has more than 300 characters long, if so
+// it'll truncate it and add an ellipsis (it will be stripped within
+// the client, but in case it isn't we might as well add it so it is
+// properly understood)
+func truncateText(original string) string {
+	if utf8.RuneCountInString(original) > 300 {
+		return fmt.Sprintf("%.300s[...]", original)
+	}
+	return original
+}
+
+func firstNImages(images []*opengraph.Image, maxImages int) []*opengraph.Image {
+	if maxImages < 0 { // dont break stuff, if it's weird, go for sane defaults
+		maxImages = MAX_IMAGES
+	}
+	numImages := len(images)
+	if numImages > maxImages {
+		subImages := make([]*opengraph.Image, maxImages)
+		subImages = images[0:maxImages]
+		return subImages
+	}
+	return images
+}
+
+// TruncateOpenGraph modifies a OG into a smaller version to ensure it
+// doesn't grow too big, as that much text won't be displayed by the
+// clients anyway. Also remove unwanted fields
+func TruncateOpenGraph(ogdata *opengraph.OpenGraph) *opengraph.OpenGraph {
+	if ogdata != nil {
+		// we might want to truncate url too, but that can have unintended effect
+		if ogdata.Title != "" {
```

Comment:
These if statements aren't necessary since they'll have no effect if the values are already empty, so we could simplify this by removing them entirely like
```go
ogdata.Title = truncateText(ogdata.Title)
ogdata.Article = nil
```

Reply:
I'll take a look, but originally ( 979c7a26a712382644b45f39c746dd0f57ae1ce2 ) it was like you mentioned and that caused a test to fail, as it were setting fields that wasn't present on the struct, so I changed it to only modify if necessary. I don't like it, as it appears to be more complex, but I felt it was necessary to be as close to the original data as I could.

Reply:
so it seems I missunderstood the error thrown from the test, so I can remove those lines and even make a more readable code (or so I hope) :) Thanks!